src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -4873,8 +4875,11 @@ void UpdateIntermediateSurfaceHeatBalanceResults(EnergyPlusData &state, ObjexxFC
             int const firstSurf = thisSpace.OpaqOrIntMassSurfaceFirst;
             int const lastSurf = thisSpace.OpaqOrIntMassSurfaceLast;
             for (int surfNum = firstSurf; surfNum <= lastSurf; ++surfNum) {
+                Real64 lowValue = 0.0000001;
                 state.dataHeatBalSurf->SurfQdotRadSolarInRepPerArea(surfNum) =
                     state.dataHeatBalSurf->SurfOpaqQRadSWInAbs(surfNum) - state.dataHeatBalSurf->SurfQdotRadLightsInPerArea(surfNum);
+                if (state.dataHeatBalSurf->SurfQdotRadSolarInRepPerArea(surfNum) <= lowValue)
+                    state.dataHeatBalSurf->SurfQdotRadSolarInRepPerArea(surfNum) = 0.0;
```

Comment:
Do we still need to add this check?

Reply:
Probably not, but I did run into some cases where there were some numerically small numbers that were getting through that were useless->like on the order of 10-8 to 10-15.  That's why I added this--to clear out the obvious junk.  If you think I should take it out, I can do that.  I'll have to alter the unit test as well since it tests the presence of this check.

Reply:
If you have a test file that shows tiny numbers, I'd like to see that. This is only zeroing out one report value, not altering anything in the heat balance, so it seems that this should be left alone to better reflect the values actually used. (I know, the separate solar and lights values aren't used in the heat balance, and the combined value isn't ever reported, so there's gonna be some small differences anyway for output vs internal values). This just adds extra steps in the heart of the simulation loop that we don't want unless there's good reason for it.

Reply:
I hear what you are saying--no need to bog things down for a report variable.  I don't remember the test file that generated this and to be honest, it was when I was testing the original solution.  I'll work to get that check removed and update the unit test.  Probably will get that wrapped up tonight or first thing tomorrow hopefully.

Reply:
Ok, I just pushed changes that eliminated the tolerance check and updated the unit test to reflect that the tolerance check was removed.  CI should run overnight; hopefully this is all set.